src/Application.vala

```diff
@@ -8,6 +8,25 @@ public class Dock.Application : Gtk.Application {
         Object (application_id: "io.elementary.dock");
     }
 
+    protected override void startup () {
+        base.startup ();
+
+        var granite_settings = Granite.Settings.get_default ();
+        var gtk_settings = Gtk.Settings.get_default ();
+
+        gtk_settings.gtk_icon_theme_name = "elementary";
+        gtk_settings.gtk_theme_name = "io.elementary.stylesheet.orange";
+
+        void check_theme () {
+            gtk_settings.gtk_application_prefer_dark_theme = (
+                granite_settings.prefers_color_scheme == Granite.Settings.ColorScheme.DARK
+            );
+        }
+
+        granite_settings.notify["prefers-color-scheme"].connect (() => check_theme ());
+        check_theme ();
```

Comment:
I don't see the reason of using a local function for a one-line. IMHO, not using it makes the code clear in this case.